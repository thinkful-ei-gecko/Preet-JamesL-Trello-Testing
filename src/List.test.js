import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
//import STORE from './STORE'
import renderer from 'react-test-renderer';


describe('<List />', () => {
    //SMOKE
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List header = "testHeader" cards = {[{title: 'test', content: 'testcontent'}, {title: 'test2', content: 'tstcnt'}]} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
});

describe('<List />', () => {
    //SCREENSHOT
    it('renders the UI as expected', () => {
        const tree = renderer.create(<List header = "testHeaderS" cards = {[{title: 'testS', content: 'testcontentS'}, {title: 'test2S', content: 'tstcntS'}]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
