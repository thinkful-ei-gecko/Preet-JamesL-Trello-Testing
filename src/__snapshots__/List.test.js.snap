// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      testHeaderS
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        testS
      </h3>
      <p>
        testcontentS
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test2S
      </h3>
      <p>
        tstcntS
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
